model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1093,6 +1093,66 @@ message SideInput {
   FunctionSpec window_mapping_fn = 3;
 }
 
+message StandardArtifacts {
+  enum Types {
+    // A URN for artifacts described by URLs.
+    // payload: ArtifactUrlPayload
+    URL      = 0 [(beam_urn) = "beam:artifact:url:v1"];
+
+    // A URN for artifacts embedded in ArtifactInformation proto.
+    // payload: EmbeddedFilePayload.
+    EMBEDDED = 1 [(beam_urn) = "beam:artifact:embedded:v1"];
+
+    // A URN for Python artifacts hosted on PYPI.
+    // payload: PypiPayload
+    PYPI     = 2 [(beam_urn) = "beam:artifact:pypi:v1"];
+
+    // A URN for Java artifacts hosted on a Maven repository.
+    // payload: MavenPayload
+    MAVEN    = 3 [(beam_urn) = "beam:artifact:maven:v1"];
+  }
+}
+
+message ArtifactUrlPayload {
+  // a string for an artifact URL e.g. "https://.../foo.jar" or "file:///tmp/foo.jar"
+  string path = 1;
+}
+
+message EmbeddedFilePayload {
+  // raw data bytes for an embedded artifact
+  bytes data = 1;
+
+  // A name of the embedded file.
+  string name = 2;
```

Comment:
This is not redundant with the top-level staged_name.

Reply:
you mean "now redundant " ?

Reply:
Yes. We can still delete it now due to the role.